src/OpenTK.Windowing.GraphicsLibraryFramework/Input/KeyboardState.cs

```diff
@@ -233,30 +169,25 @@ public override string ToString()
             return builder.ToString();
         }
 
-        // This shouldn't be necessary but I'll keep it in just in case.
-        [Conditional("DEBUG")]
-        private static void ValidateOffset(int offset)
+        internal void Update()
         {
-            if (offset < 0 || offset >= NumInts * IntSize)
-            {
-                throw new ArgumentOutOfRangeException();
-            }
+	        Utils.Swap(ref _keys, ref _keysPrevious);
         }
 
-        private static (int intOffset, int bitOffset) GetOffsets(Key key)
+        /// <summary>
+        /// Gets a <see cref="bool" /> indicating whether this key was down in the previous frame.
+        /// </summary>
+        /// <param name="key">The <see cref="Key" /> to check.</param>
+        /// <returns><c>true</c> if <paramref name="key"/> was in the down state; otherwise, <c>false</c>.</returns>
+        public bool WasKeyDown(Keys key)
         {
-            if (key <= Key.Unknown || key > Key.LastKey)
-            {
-                throw new ArgumentOutOfRangeException(nameof(key), "Invalid key");
-            }
-
-            var offset = (int)key;
-            ValidateOffset(offset);
-
-            var intOffset = offset / IntSize;
-            var bitOffset = offset % IntSize;
-
-            return (intOffset, bitOffset);
+	        return _keysPrevious[(int)key];
         }
+
+        /// <summary>
+        /// Gets an immutable snapshot of this KeyboardState.
```

Comment:
We could improve the docstring by explaining what the method is for: "This can be used to save the current keyboard state for comparison or reference later on."